.github/workflows/add_ported_warnings.yml

```diff
@@ -25,5 +25,30 @@ jobs:
         uses: Ana06/get-changed-files@v1.2
 
       - name: run the script
+        id: script
         run: |
           python scripts/detect_ported_files.py ${{ steps.changed-files.outputs.all }}
+
+      - id: PR
+        uses: 8BitJonny/gh-get-current-pr@2.2.0
+        # TODO: this may not work properly if the same commit is pushed to multiple branches:
+        # https://github.com/8BitJonny/gh-get-current-pr/issues/8
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          sha: ${{ github.event.pull_request.head.sha }}
+          # Only return if PR is still open
+          filterOutClosed: true
+
+      - if: steps.script.outputs.modifies_ported == 'True'
+        id: add_label
+        name: add "modifies-synchronized-file"
+        # we use curl rather than octokit/request-action so that the job won't fail
+        # (and send an annoying email) if the labels don't exist
```

Comment:
I would guess that a `github-script` action would be easier to manage, and then you can try/catch in javascript.

Reply:
I mean this is just copy pasted from the other file

Reply:
If you want to refactor it all go ahead but I've spent way too much time fixing up gh actions recently

Reply:
I'm not sure if the comment really makes sense, the reason I use curl is because I got an error I didn't understand with the other version using actions properly. Which may now be fixed by the sha line who knows, but this whole setup should only be relevant for the next couple of months so I wasn't too worried about it being a hack